include/openssl/ssl.h

```diff
@@ -1395,6 +1395,31 @@ __owur int SSL_CIPHER_get_bits(const SSL_CIPHER *c, int *alg_bits);
 __owur char *SSL_CIPHER_get_version(const SSL_CIPHER *c);
 __owur const char *SSL_CIPHER_get_name(const SSL_CIPHER *c);
 __owur uint32_t SSL_CIPHER_get_id(const SSL_CIPHER *c);
+typedef enum {
+    SSL_CIPHER_KX_RSA = 1,
+    SSL_CIPHER_KX_ECDHE,
+    SSL_CIPHER_KX_DHE,
+    SSL_CIPHER_KX_ECDHE_PSK,
+    SSL_CIPHER_KX_DHE_PSK,
+    SSL_CIPHER_KX_RSA_PSK,
+    SSL_CIPHER_KX_PSK,
+    SSL_CIPHER_KX_SRP,
+    SSL_CIPHER_KX_GOST
```

Comment:
Ok, the proposed solution for enum extensibility is:

1) Choose a value that is larger than the number of enum values ever expected (during the lifetime of a compatible branch). For KX and auth, I'd think that 255 is sufficient.
2) Define an SSL_CIPHER_KX_PRIVATE_RESERVED_FOR_EXPANSION_DO_NOT_USE with that value. (Name deliberately obnoxious.)
3) Add a comment saying
```
/*
 * New enum values may be added in compatible releases.
 * Applications must be prepared to handle unknown enum values.
 * SSL_CIPHER_KX_PRIVATE_RESERVED_FOR_EXPANSION_DO_NOT_USE
 * is used to reserve the enum range, and to aid compilers (for example, to catch
 * a missing 'default' in a switch statement.
 * Applications MUST NOT reference this value explicitly. Instead, they should handle
 * an unknown default when necessary.
 */
```
4) Document the fact that enums may be extensible; don't document the private value.

Does this make sense?